### PR TITLE
Adding an additional popover closing method

### PR DESCRIPTION
**Pull request by ollips**
Task: task-[work-item-number]

Link: preview-326

Adding the `data-close` attribute to the popover's main content link/button would collapse it after a click.

## Testing

1. Step one
2. Step two
   Expected result:

## Additional information

[Optional]

### CODE REVIEW

File: site/src/components/popover.md [hunk 1]
```diff
@@ -71,3 +71,26 @@ Popover is left aligned by default, but also can be centered or right aligned re
 	<div class="popover-content border">Popover content.</div>
 </details>
 ```
+
+## Closing popover
+
+An additional way to collapse the popover is to add the `data-close` attribute to the link/button in the content.
```
Comment (wibjorn) [inline site/src/components/popover.md:77]: I think we should name scope this to popovers `data-popover-close`?

File: site/src/components/popover.md [hunk 2]
```diff
@@ -71,3 +71,26 @@ Popover is left aligned by default, but also can be centered or right aligned re
...
+	<div class="popover-content border">
+		<ul>
+			<li class="margin-bottom-xxs">
+				<a href="#">Link</a>
+			</li>
+			<li class="margin-bottom-xxs">
+				<a href="#">Link</a>
+			</li>
+			<li>
+				<a href="#" data-close>Closing link</a>
```
Comment (wibjorn) [inline site/src/components/popover.md:91]: Perhaps the example should be a button?

File: site/src/components/popover.md [hunk 3]
```diff
@@ -71,3 +71,26 @@ Popover is left aligned by default, but also can be centered or right aligned re
 	<div class="popover-content border">Popover content.</div>
 </details>
 ```
+
```
Comment (wibjorn) [inline site/src/components/popover.md:74]: Wondering if we can omit this documentation until or re-write it to be a suggested implementation. We don't yet have a way of sharing the front end scripts with other consumers (we're documenting TS that is not technically part of our css package)

File: site/src/scaffold/scripts/popover.ts [hunk 4]
```diff
@@ -28,6 +28,10 @@ export function initPopovers(container: HTMLElement) {
 				if (!targetPopover?.contains(event.target)) {
 					closePopovers();
 				}
+
+				if (event.type === 'click' && event.target.hasAttribute('data-close')) {
```
Comment (wibjorn) [inline site/src/scaffold/scripts/popover.ts:32]: `data-popover-close`?

File: site/src/components/popover.md (hunk 1 above)
Comment (ollips) [reply site/src/components/popover.md]: Updated

File: site/src/components/popover.md (hunk 3 above)
Comment (ollips) [reply site/src/components/popover.md]: Oh, you're right, I always forget we don't share the scripts right now. Decided to omit this portion of documentation until we do!

File: site/src/components/popover.md (hunk 2 above)
Comment (ollips) [reply site/src/components/popover.md]: Temporarily removed the whole example block

File: site/src/scaffold/scripts/popover.ts (hunk 4 above)
Comment (ollips) [reply site/src/scaffold/scripts/popover.ts]: Updated

**Review by wibjorn — APPROVED** [review]

**Review by homing1 — APPROVED** [review]

**Pull request merged**